languages/csharp-dotnet-2.2/template/src/__ALGO__.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using Algorithmia;
+
+//This is an example of an algorithm, you can evaluate it with this input:
+/// Example Input:
+/// {"name": "Algorithmia user"}
+///
+/// Expected Output:
+/// {"output":"Hello Algorithmia user"}
+namespace __ALGO__
```

Comment:
Random: as we look forward to features like forking, I wonder if keeping `__ALGO__` or `__USER__` in the source code (or filenames) for IPA is a good strategy? It might be too late, cuz I imagine the python stuff already does this, but I feel like this is one of those tiny decisions that affects our ability to think about features like forking.

Reply:
Python does not,  we parse the algorithmia.conf file https://github.com/algorithmiaio/langpacks/blob/master/libraries/python3-runtime/context/pipe#L15

And I also agree with you it's a major decision

Reply:
As this algorithm is a compiled language, you can't parse the algorithmia.conf file without some kind of runtime reflection. Python is a scripting language which allows you to import dynamically without any design decisions at all. Rust would normally run into this same issue, but I believe with how we do the cargo install, we read from the algorithmia.conf file in advance using a secondary module. I have no idea how to do do this for C#, or whether it's even possible.

We had a conversation about this a few times, specifically dynamic loading of packages using something like a DLL loader (which makes this easy) vs. static compilation (which makes this hard). and it seemed like static compilation won - but maybe there was a misfire in communication somewhere?

What would you like to go folks?

Reply:
We've simplified the problem by removing the file renaming system entirely. From now on we use a generic `Algorithm` namespace, class and filename. This removes issues we may run into with forking and makes things consistent with the Rust client.